iOS link error with stbimage

**Undefined symbol of stbi_load, stbi_info**
Undefined symbols for architecture arm64:
  "_stbi_load", referenced from:
      gltfio::ResourceLoader::createTextures(gltfio::details::FFilamentAsset*) const::$_4::operator()() const in libgltfio_core.a(ResourceLoader.cpp.o)
     (maybe you meant: _stbi_loadf_from_callbacks, _stbi_loadf_from_memory , _stbi_load_from_callbacks , _stbi_load_16_from_callbacks , _stbi_load_16_from_memory , _stbi_load_from_memory )
  "_stbi_info", referenced from:
      gltfio::ResourceLoader::createTextures(gltfio::details::FFilamentAsset*) const in libgltfio_core.a(ResourceLoader.cpp.o)
     (maybe you meant: _stbi_info_from_callbacks, _stbi_info_from_memory )
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**To Reproduce**
run ios gltf sample.

**Screenshots**
![image](https://user-images.githubusercontent.com/24448050/73289907-c5d7b180-41fd-11ea-9097-c03bb025b346.png)

--- a/libs/gltfio/include/gltfio/Image.h
+++ b/libs/gltfio/include/gltfio/Image.h
@@ -22,6 +22,11 @@
 #define STBI_NO_HDR
 #define STBI_NO_PIC
 #define STBI_NO_PNM
+
+// For emscripten and Android builds, we never load from the file
+// system, so we-opt out of the stdio functionality in stb.
+#if defined(__EMSCRIPTEN__) || defined(ANDROID)
 #define STBI_NO_STDIO
+#endif
 
 #include <stb_image.h>

--- a/libs/gltfio/src/ResourceLoader.cpp
+++ b/libs/gltfio/src/ResourceLoader.cpp
@@ -15,6 +15,7 @@
  */
 
 #include <gltfio/ResourceLoader.h>
+#include <gltfio/Image.h>
 
 #include "FFilamentAsset.h"
 #include "upcast.h"
@@ -35,8 +36,6 @@
 #include <utils/Log.h>
 
 #include <cgltf.h>
-
-#include <stb_image.h>
 
 #include <tsl/robin_map.h>
 
@@ -147,7 +146,7 @@ bool ResourceLoader::loadResources(FilamentAsset* asset) {
     // looks inside a cache of externally-supplied data blobs, rather than loading from the
     // filesystem.
 
-    #if defined(__EMSCRIPTEN__) || defined(ANDROID)
+    #if defined(STBI_NO_STDIO)
 
     if (gltf->buffers_count && !gltf->buffers[0].data && !gltf->buffers[0].uri && gltf->bin) {
         if (gltf->bin_size < gltf->buffers[0].size) {
@@ -378,7 +377,7 @@ bool ResourceLoader::createTextures(details::FFilamentAsset* asset) const {
                 js->run(decode);
             }
         } else {
-            #if defined(__EMSCRIPTEN__) || defined(ANDROID)
+            #if defined(STBI_NO_STDIO)
                 slog.e << "Unable to load texture: " << tb.uri << io::endl;
                 return false;
             #else

